src/fidesops/ops/service/saas_request/override_implementations/firebase_auth_request_overrides.py

```diff
@@ -0,0 +1,155 @@
+from fidesops.ops.models.policy import Policy
+from fidesops.ops.models.privacy_request import PrivacyRequest
+from fidesops.ops.service.saas_request.saas_request_override_factory import (
+    SaaSRequestType,
+    register,
+)
+from fidesops.ops.util.collection_util import Row
+
+logger = logging.getLogger(__name__)
+
+
+@register("firebase_auth_user_access", [SaaSRequestType.READ])
+def firebase_auth_user_access(
+    node: TraversalNode,
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    input_data: Dict[str, List[Any]],
+    secrets: Dict[str, Any],
+) -> List[Row]:
+    """
+    SaaS Request Override function used to integrate access requests with
+    Firebase Auth service. Specifically, it reads `User` records.
+    """
+    app = initialize_firebase(secrets)
+
+    processed_data = []
+
+    emails = input_data.get("email", [])
+    for email in emails:
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        row = {"email": user.email, "uid": user.uid}
+        if user.display_name is not None:
+            row["display_name"] = user.display_name
+        if user.phone_number is not None:
+            row["phone_number"] = user.phone_number
+        if user.disabled is not None:
+            row["disabled"] = user.disabled
+        if user.email_verified is not None:
+            row["email_verified"] = user.email_verified
+        if user.provider_data:
+            pds = []
+            for pd in user.provider_data:
+                pds.append(
+                    {
+                        "display_name": pd.display_name,
+                        "provider_id": pd.provider_id,
+                        "email": pd.email,
+                    }
+                )
+            row["provider_data"] = pds
+        processed_data.append(row)
+
+    return processed_data
+
+
+@register("firebase_auth_user_update", [SaaSRequestType.UPDATE])
+def firebase_auth_user_update(
+    param_values_per_row: List[Dict[str, Any]],
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    secrets: Dict[str, Any],
+) -> int:
+    """
+    SaaS Request Override function used to integrate update requests with
+    Firebase Auth service. Specifically, it updates `User` records.
+    """
+    app = initialize_firebase(secrets)
+    rows_updated = 0
+    # each update_params dict correspond to a record that needs to be updated
+    for row_param_values in param_values_per_row:
+        email = row_param_values.get("email")
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        masked_fields = row_param_values["masked_object_fields"]
+        display_name = masked_fields.get("display_name", user.display_name)
+        phone_number = masked_fields.get("phone_number", user.phone_number)
+        disabled = masked_fields.get("disabled", user.disabled)
+        email_verified = masked_fields.get("email_verified", user.email_verified)
+
+        auth.update_user(
```

Comment:
`provider_data` cannot be updated programmatically via the API, so we don't try to update it here.

Reply:
that's not a bad point. perhaps it may depend on the use case - maybe we need some input from our first use case, and generally we should look to enable both pieces of functionality (as i've done by defining `update` and `delete` override functions)

Reply:
we're going to try to include some test coverage for both update and delete functionality, and keep both override functions in the codebase. that way it will essentially be a config switch (i.e. updating the saas config) whether to use update or delete in any given use case. we can adjust the default if needed, once we get some feedback from the real-world use cases.